src/LibraryManager/Providers/Cdnjs/CdnjsProvider.cs

```diff
@@ -319,35 +311,35 @@ public string GetLibraryVersion(string libraryId)
             return LibraryOperationResult.FromSuccess(state);
         }
 
-        private bool IsLibraryUpToDateAsync(ILibraryInstallationState state, CancellationToken cancellationToken)
+        private bool IsLibraryUpToDate(ILibraryInstallationState state)
         {
-            string name = GetLibraryName(state.LibraryId);
-            string version = GetLibraryVersion(state.LibraryId);
-
-            string cacheDir = Path.Combine(CacheFolder, name, version);
-            string destinationDir = Path.Combine(HostInteraction.WorkingDirectory, state.DestinationPath);
-
             try
             {
-                foreach (string sourceFile in state.Files)
+                (string name, string version) = LibraryNamingScheme.Instance.GetLibraryNameAndVersion(state.LibraryId);
+
+                if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(version))
```

Comment:
>if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(version)) [](start = 16, length = 66)

design follow up: Can we avoid doing this check in many places?

Reply:
Logged issue 252

---
In reply to: [197970963](https://github.com/aspnet/LibraryManager/pull/230#discussion_r197970963) [](ancestors = 197970963)